File: cases/taint/scala/map-iteration-positive/MapIterationPositive.scala
package dataflowbench

object MapIterationPositive {
  def dfb_source(): String = { // DFB-SOURCE: map-iteration-input
    "tainted"
  }

  def dfb_sink(value: String): Unit = {} // DFB-SINK: map-iteration-sink

  def run(): Unit = {
    val records = scala.collection.mutable.Map[String, String]()
    records("record") = dfb_source() // DFB-WITNESS: map-iteration-store
    for ((key, value) <- records) {
      dfb_sink(value)
    }
  }
}

File: cases/taint/kotlin/map-iteration-positive/MapIterationPositive.kt
package dataflowbench

object MapIterationPositive {
    fun dfb_source(): String { // DFB-SOURCE: map-iteration-input
        return "tainted"
    }

    fun dfb_sink(value: String) {} // DFB-SINK: map-iteration-sink

    fun run() {
        val records = mutableMapOf<String, String>()
        records["record"] = dfb_source() // DFB-WITNESS: map-iteration-store
        for ((key, value) in records) {
            dfb_sink(value)
        }
    }
}
